src/auth/app/service.go

```diff
@@ -3,112 +3,72 @@ package app
 import (
 	"context"
 	"crypto/x509"
-
-	"github.com/consensys/quorum-key-manager/src/auth/api/middlewares"
-	"github.com/consensys/quorum-key-manager/src/auth/entities"
-	"github.com/consensys/quorum-key-manager/src/auth/manager"
+	"github.com/consensys/quorum-key-manager/pkg/app"
+	service "github.com/consensys/quorum-key-manager/src/auth"
+	"github.com/consensys/quorum-key-manager/src/auth/api/http_middlewares"
+	"github.com/consensys/quorum-key-manager/src/auth/api/manifest"
+	authtypes "github.com/consensys/quorum-key-manager/src/auth/entities"
 	"github.com/consensys/quorum-key-manager/src/auth/service/authenticator"
-	apikey "github.com/consensys/quorum-key-manager/src/infra/api-key/filesystem"
+	"github.com/consensys/quorum-key-manager/src/auth/service/roles"
+	entities2 "github.com/consensys/quorum-key-manager/src/entities"
 	"github.com/consensys/quorum-key-manager/src/infra/jwt"
-	"github.com/consensys/quorum-key-manager/src/infra/jwt/jose"
 	"github.com/consensys/quorum-key-manager/src/infra/log"
-	manifestreader "github.com/consensys/quorum-key-manager/src/infra/manifests/filesystem"
-	tls "github.com/consensys/quorum-key-manager/src/infra/tls/filesystem"
 	"github.com/justinas/alice"
-
-	"github.com/consensys/quorum-key-manager/pkg/app"
 )
 
-func RegisterService(a *app.App, logger log.Logger) error {
-	// Load configuration
-	cfg := new(Config)
-	err := a.ServiceConfig(cfg)
-	if err != nil {
-		return err
+func RegisterService(
+	ctx context.Context,
+	a *app.App,
+	logger log.Logger,
+	manifests map[string][]entities2.Manifest,
+	jwtValidator jwt.Validator,
+	apikeyClaims map[string]*authtypes.UserClaims,
+	rootCAs *x509.CertPool,
+) (*roles.Interactor, error) {
+	// Business layer
+	// TODO: Create authorizator service here
+
+	var authenticatorService *authenticator.Authenticator
+	if jwtValidator != nil || apikeyClaims != nil || rootCAs != nil {
```

Comment:
Having services which might be nil depending on the settings of the user appears to be a bad practices since it forces us to do checks over our service execution flow. I would rather encapsulate the fact not authentication methods were enabled under the authService itself